### PR TITLE
Add support for other RP2040-based boards

### PR DESCRIPTION
**Pull request by smittytone**
Here is the restored PR. It now builds via the command line with `cmake` *and* through VSCode, with the former taking the user’s board choice from `PICO_BOARD` if defined and exported. Or folk can set this in `CmakeLists.txt`, which is probably the route we VSCode users will take, but whatever. 

This PR targets `master`, but I recommend creating a new branch — I'll re-target the PR to that. That way you can merge it there, make whatever changes you see fit, and merge to `master` in due course if you wish.

Tested on the Pico, QTPy RP2040, Pro Micro RP2040 and Feather RP2040 on Mac, built at the command line and under VSCode. 

There remains the issue of `PICO_DEFAULT_LED_PIN not defined` errors with non-Pico builds, cludged in my code. I was thinking of adding this to `CMakeLists.txt` before the first `add_executable()` is called (and `picoprobe_config.h` processed), but haven't had time to test it.

```
if (NOT ${PICO_BOARD} STREQUAL "pico")
    add_compile_definitions(PICO_DEFAULT_LED_PIN=0)
endif()
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Picoprobe
+
+A Picoprobe is a Raspberry Pi RP2040-based Serial Wire Debug (SWB) adaptor. You can make one from a Raspberry Pi Pico — hence the name — or almost any third-party RP2040-based board. Use it to connect your RP2040-based application board to your personal computer for debugging through `gdb`, directly or via an IDE such as Microsoft Visual Studio Code.
+
+This code originally supported only the Pico. It now supports three more boards, each of which provides a [STEMMA QT connector](https://learn.adafruit.com/introducing-adafruit-stemma-qt/what-is-stemma-qt)/[Qwiic](https://www.sparkfun.com/qwiic) for easy cabling to the application board under test. These connectors are intended for use with I&sup2;C devices, but here we’re using their pins for GPIO and power.
+
+![A Picoprorbe build from the Adafruit QTPy RP2040](images/qtpy_picoprobe.jpg)
```
Comment (lurch) [inline README.md:7]: Picoprorbe -> Picoprobe ? :wink:

File: README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Picoprobe
+
+A Picoprobe is a Raspberry Pi RP2040-based Serial Wire Debug (SWB) adaptor. You can make one from a Raspberry Pi Pico — hence the name — or almost any third-party RP2040-based board. Use it to connect your RP2040-based application board to your personal computer for debugging through `gdb`, directly or via an IDE such as Microsoft Visual Studio Code.
```
Comment (lurch) [inline README.md:3]: SWB -> SWD

**Comment by lurch** [comment]
> There remains the issue of PICO_DEFAULT_LED_PIN not defined errors with non-Pico builds, cludged in my code.

Did you try the `#if !defined(PICO_DEFAULT_LED_PIN) && !defined(PICO_DEFAULT_WS2812_PIN)` that I suggested in your [other PR](https://github.com/raspberrypi/picoprobe/pull/18) ?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,31 +1,81 @@
...
+elseif (${PICO_BOARD} STREQUAL "adafruit_feather_rp2040")
+    message("Building for Adafruit Feather RP2040 with WS2812")
+elseif (${PICO_BOARD} STREQUAL "sparkfun_promicro")
+    message("Building for SparkFun Pro Micro RP2040 with WS2812")
+else()
+    message("Building for Raspberry Pi Pico")
+endif()
+
+if (NOT ${PICO_BOARD} STREQUAL "pico")
+    target_sources(${EXECUTABLE_NAME} PRIVATE src/ws2812.c)
```
Comment (lurch) [inline CMakeLists.txt:57]: This seems to be assuming that any board that *isn't* a Raspberry Pi Pico _does_ have a WS2812 LED, which doesn't feel like a very flexible / easily-extendable solution? :man_shrugging:

**Comment by smittytone** [comment]
SDK changes in the interim have made this PR redundant, I think, so I plan to close it unless I hear otherwise.

**Pull request closed without merging**